torch/_dynamo/polyfill.py

```diff
@@ -63,6 +63,31 @@ def set_isdisjoint(set1, set2):
     return True
 
 
+def set_intersection(set1, set2):
+    intersection_set = set()
+    for x in set1:
+        if x in set2:
+            intersection_set.add(x)
+    return intersection_set
+
+
+def set_union(set1, set2):
+    union_set = set()
+    for s in (set1, set2):
+        for x in s:
+            if x not in union_set:
+                union_set.add(x)
+    return union_set
```

Comment:
```suggestion
def set_union(set1, set2):
    union_set = set1.copy()
    for x in set2:
        if x not in union_set:
            union_set.add(x)
    return union_set
```

Reply:
@XuehaiPan Thank you very much for your suggestions! I have incorporated your suggestions. Could you please review the PR again? cc @Skylion007 @anijain2305